upup/pkg/fi/cloudup/openstack/cloud.go

```diff
@@ -62,3 +127,78 @@ func (c *openstackCloud) DeleteGroup(g *cloudinstances.CloudInstanceGroup) error
 func (c *openstackCloud) GetCloudGroups(cluster *kops.Cluster, instancegroups []*kops.InstanceGroup, warnUnmatched bool, nodes []v1.Node) (map[string]*cloudinstances.CloudInstanceGroup, error) {
 	return nil, fmt.Errorf("openstackCloud::GetCloudGroups not implemented")
 }
+
+func (c *openstackCloud) SetVolumeTags(id string, tags map[string]string) error {
+	if len(tags) == 0 {
+		return nil
+	}
+	if id == "" {
+		return fmt.Errorf("error setting tags to unknown volume")
+	}
+	glog.V(4).Infof("setting tags to cinder volume %q: %v", id, tags)
+
+	opt := cinder.UpdateOpts{Metadata: tags}
+	done, err := vfs.RetryWithBackoff(writeBackoff, func() (bool, error) {
```

Comment:
We should probably pull this into a `retry` package if we're going to be using it throughout the code, but not a problem...